pkg/kwokctl/runtime/scheduler.go

```diff
@@ -18,21 +18,41 @@ package runtime
 
 import (
 	"fmt"
+	"os"
+
+	v1 "k8s.io/kube-scheduler/config/v1"
+	"sigs.k8s.io/kwok/pkg/utils/yaml"
 )
 
 // CopySchedulerConfig copies the scheduler configuration file to the given path.
 func (c *Cluster) CopySchedulerConfig(oldpath, newpath, kubeconfig string) error {
-	err := c.CopyFile(oldpath, newpath)
+	var config v1.KubeSchedulerConfiguration
+
+	if !c.dryRun {
+		data, err := os.ReadFile(oldpath)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %w", oldpath, err)
+		}
+
+		err = yaml.Unmarshal(data, &config)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal YAML from %s: %w", oldpath, err)
+		}
+
```

Comment:
Add validations ensuring that the "apiVersion" is set to `v1.SchemeGroupVersion.String()` and that the "kind" is set to "KubeSchedulerConfiguration"